### PR TITLE
customizable main menu without html

### PR DESCRIPTION
**Pull request by xithan**

### CODE REVIEW

File: card/spec/lib/card/content_spec.rb [hunk 1]
```diff
@@ -191,11 +191,15 @@
       @card = card
 
       # non-nil valued opts only ...
-      @render_block = proc do |opts|
-        options = opts.inject({}) do |i, v|
-          i if !v[1].nil? && (i[v[0]] = v[1])
+      @render_block = proc do |chunk|
```
Comment (ethn) [inline card/spec/lib/card/content_spec.rb]: this is harder to understand than the actual code.

File: card/spec/support/helper/view_helper.rb [hunk 2]
```diff
@@ -40,7 +40,17 @@ def render_card_with_args view, card_args={}, format_args={}, view_args={}
...
-            Card.fetch card_args[:name], new: card_args
+            if Card.real? card_args[:name]
+              card = Card.fetch card_args.delete(:name)
+              card_args.each do |k, v|
+                card.send "#{k}=", v
+              end
+              card
+            else
+              Card.fetch card_args[:name], new: card_args
+            end
```
Comment (ethn) [inline card/spec/support/helper/view_helper.rb]: give this its own method?

File: card/mod/navbar/set/all/navbar_links.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
+format :html do
+  view :navbar_links, perms: :none do
+    wrap_with :ul, class: "navbar-nav" do
+      navbar_items
+    end
+  end
+
+  def navbar_items view: :nav_item, link_class: "nav-link"
```
Comment (ethn) [inline card/mod/navbar/set/all/navbar_links.rb]: This is cool.  It does enough work that it's probably worth a comment.

File: card/lib/card/format/content.rb [hunk 4]
```diff
@@ -1,15 +1,21 @@
...
         content = override_content || render_raw || ""
         content_object = get_content_object content, content_opts
-        content_object.process_each_chunk do |chunk_opts|
-          content_nest chunk_opts
-        end
+        content_object.process_chunks(&block)
         content_object.to_s
       end
 
+      def safe_process_content override_content=nil, content_opts=nil, &block
```
Comment (ethn) [inline card/lib/card/format/content.rb]: This should probably have a short comment to explain "safe" to our later selves.

File: card/lib/card/content/chunk.rb [hunk 5]
```diff
@@ -140,15 +140,27 @@ def to_s
         end
 
         def result
-          @process_chunk || @processed || @text
+          burn_read || @process_chunk || @processed || @text
+        end
+
+        def burn_read
```
Comment (ethn) [inline card/lib/card/content/chunk.rb]: short comment?

File: card/db/migrate_core_cards/data/cards/Xmain_menu [hunk 6]
```diff
@@ -1,2 +1,2 @@
-<a class="nav-link internal-link" href="/">Home</a>
-<a class="nav-link internal-link" href="/:recent">Recent</a>
\ No newline at end of file
+[[*getting started+links|Getting started]]
+[[:recent|Recent]]
```
Comment (ethn) [inline card/db/migrate_core_cards/data/cards/Xmain_menu]: 😍

File: card/spec/lib/card/content_spec.rb (hunk 1 above)
Comment (xithan) [reply card/spec/lib/card/content_spec.rb]: The only goal was to get the tests pass again. Make a story for it with at least 2 points then I'll make it beautiful :sunglasses:

**Pull request merged**